pyethapp/tests/test_console_service.py

```diff
@@ -0,0 +1,146 @@
+from itertools import count
+import pytest
+import serpent
+from devp2p.peermanager import PeerManager
+import ethereum
+from ethereum import tester
+from ethereum.ethpow import mine
+import ethereum.keys
+import ethereum.config
+from ethereum.slogging import get_logger
+from pyethapp.accounts import Account, AccountsService, mk_random_privkey
+from pyethapp.app import EthApp
+from pyethapp.config import update_config_with_defaults, get_default_config
+from pyethapp.db_service import DBService
+from pyethapp.eth_service import ChainService
+from pyethapp.jsonrpc import JSONRPCServer
+from pyethapp.pow_service import PoWService
+from pyethapp.console_service import Console
+
+# reduce key derivation iterations
+ethereum.keys.PBKDF2_CONSTANTS['c'] = 100
+
+
+log = get_logger('test.console_service')
+
+
+@pytest.fixture
+def test_app(request, tmpdir):
+
+    class TestApp(EthApp):
+
+        def start(self):
+            super(TestApp, self).start()
+            log.debug('adding test accounts')
+            # high balance account
+            self.services.accounts.add_account(Account.new('', tester.keys[0]), store=False)
+            # low balance account
+            self.services.accounts.add_account(Account.new('', tester.keys[1]), store=False)
+            # locked account
+            locked_account = Account.new('', tester.keys[2])
+            locked_account.lock()
+            self.services.accounts.add_account(locked_account, store=False)
+            assert set(acct.address for acct in self.services.accounts) == set(tester.accounts[:3])
+
+        def mine_next_block(self):
+            """Mine until a valid nonce is found.
+
+            :returns: the new head
+            """
+            log.debug('mining next block')
+            block = self.services.chain.chain.head_candidate
+            delta_nonce = 10**6
+            for start_nonce in count(0, delta_nonce):
+                bin_nonce, mixhash = mine(block.number, block.difficulty, block.mining_hash,
+                                          start_nonce=start_nonce, rounds=delta_nonce)
+                if bin_nonce:
+                    break
+            self.services.pow.recv_found_nonce(bin_nonce, mixhash, block.mining_hash)
+            log.debug('block mined')
+            assert self.services.chain.chain.head.difficulty == 1
+            return self.services.chain.chain.head
+
+        def rpc_request(self, method, *args):
```

Comment:
hi @konradkonrad - I believe I probably copied over more than I should have from the pyethapp/tests/test_jsonrpc.py file.   I used that as a template to get started on the console service tests.   I'll get that cleaned up this evening and update the PR.   thank you for pointing that out

Brian